File: jeweltilt/craneam/com/Jewel.java
package jeweltilt.craneam.com;

import android.util.Log;

public class Jewel
  extends Quad
  {
    private static final String TAG = "Jewel Tilt";  // Game tag in logs
    private static final boolean DEBUG = false;
    private int scoreValue;
    private int type;
    private float[] coords = new float[3];  // current position

    public Jewel(float size, int points, int jewelType, float scale)
      {
        super(size, scale);
        scoreValue = points;
        type = jewelType;
      }

    public int GetType()
      {
        return type;
      }

    public int GetScoreValue()
      {
        return scoreValue;
      }

    // Function to set position
    public void SetCoordinates(float x, float y, float z)
      {
        coords[0] = x;
        coords[1] = y;
        coords[2] = z;
        if (DEBUG)
          {
            Log.d(TAG, "Jewel coordinates set to: " + x + ", " + y + ", " + z);
          }
      }

    // Functions to return coordinates
    public float GetXCoordinate()
      {
        return coords[0];
      }

    public float GetYCoordinate()
      {
        return coords[1];
      }

    public float GetZCoordinate()
      {
        return coords[2];
      }
  }

File: app/src/main/java/com/koodipuukko/twintilt/Jewel.kt
package com.koodipuukko.twintilt

import android.util.Log

class Jewel
	(
	size: Float,
	private val scoreValue: Int,
	private val type: Int,
	scale: Float
) : Quad(size, scale) {
	private val coords = FloatArray(3) // current position
	fun getType(): Int {
		return type
	}

	fun getScoreValue(): Int {
		return scoreValue
	}
	// Function to set position
	fun setCoordinates(x: Float, y: Float, z: Float) {
		coords[0] = x
		coords[1] = y
		coords[2] = z
		if(DEBUG) {
			Log.d(TAG, "Jewel coordinates set to: $x, $y, $z")
		}
	}
	// Functions to return coordinates
	fun getXCoordinate(): Float {
		return coords[0]
	}

	fun getYCoordinate(): Float {
		return coords[1]
	}

	fun getZCoordinate(): Float {
		return coords[2]
	}

	companion object {
		private const val TAG = "Twintilt" // Game tag in logs
		private const val DEBUG = false
	}
}